src/lib/viewers/controls/media/MediaSettingsControls/MediaSettingsFlyout.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Menu, Ref as MediaSettingsMenuRef } from './MediaSettingsMenu';
+import './MediaSettingsFlyout.scss';
+
+export type Props = {
+    children: React.ReactNode;
+    className?: string;
+    menu: Menu;
+};
+
+const SETTINGS_MENU_PADDING = 18;
+const SETTINGS_MENU_PADDING_SCROLL = 32;
+const SETTINGS_MENU_MAX_HEIGHT = 210;
+
+export default function MediaSettingsFlyout({ children, className, menu }: Props): JSX.Element | null {
+    const [dimension, setDimension] = React.useState({});
+    const [isInTransition, setIsInTransition] = React.useState(false);
+    const refs = React.useRef<Array<MediaSettingsMenuRef>>([]);
+
+    const handleTransitionEnd = (): void => setIsInTransition(false);
+
+    React.useEffect(() => {
+        const nextMenuEl = refs.current.find(ref => ref.classList.contains('bp-is-active'));
+
+        if (nextMenuEl) {
```

Comment:
Instead of the array of refs, would it work if we just triggered the effect when menu value changes?